tests/unit/mixins/route-test.js

```diff
@@ -150,86 +152,78 @@ test('it can not use infinityModel without a Model Name', assert => {
-  var model = callModelHook(route);
+  let model = callModelHook(route);
 
-  assert.equal(route.get('_totalPages'), 31, '_totalPages');
-  assert.equal(route.get('currentPage'), 1, 'currentPage');
-  assert.equal(route.get('_canLoadMore'), true, '_canLoadMore');
+  assert.equal(model.get('_totalPages'), 31, '_totalPages');
+  assert.equal(model.get('currentPage'), 1, 'currentPage');
+  assert.equal(model.get('_canLoadMore'), true, '_canLoadMore');
   assert.ok(Ember.$.isEmptyObject(route.get('_extraParams')), 'extra params are empty');
   assert.ok(!model.get('reachedInfinity'), 'Should not reach infinity');
 });
 
 test('it allows customizations of request params', assert => {
-  var store = createMockStore(
-    Ember.Object.create({ items: [] }),
-    function (modelType, findQuery) {
-      assert.deepEqual(findQuery, {per: 25, p: 1}, 'findQuery');
-  });
+  let store = createMockStore(
+    EA([]), 
+    (modelType, findQuery) => {
+      assert.deepEqual(findQuery, { per: 25, p: 1 }, 'findQuery');
+    }
+  );
 
-  var route = createRoute(['item'], {
-    perPageParam: 'per',
-    pageParam: 'p',
-    store
-  });
+  let route = createRoute(['item', { 
+    perPageParam: 'per', pageParam: 'p'
+  }], { store });
 
   callModelHook(route);
 });
 
+test('It allows to set startingPage as 0', assert => {
+  let store = createMockStore( EA([{id: 1, name: 'Test'}], { total_pages: 1 }) );
+  let route = createRoute(['item', {
+    startingPage: 0
+  }], { store });
+
+  let model = callModelHook(route);
+
+  assert.equal(route.get('currentPage'), 0);
+  assert.equal(model.get('_canLoadMore'), false);
+});
+
 test('it skips request params when set to null', assert => {
-  var store = createMockStore(
-    Ember.Object.create({ items: [] }),
-    function (modelType, findQuery) {
+  let store = createMockStore(
+    EA([]), 
+    (modelType, findQuery) => {
       assert.deepEqual(findQuery, {}, 'findQuery');
   });
 
-  var route = createRoute(['item'], {
-    perPageParam: null,
-    pageParam: null,
-    store
-  });
+  let route = createRoute(['item', { 
+    perPageParam: null, pageParam: null
+  }], { store });
 
   callModelHook(route);
 });
 
-
 test('it allows customizations of meta parsing params', assert => {
-  var store = createMockStore(Ember.Object.create({
-    items: [{id: 1, name: 'Walter White'}],
-    pagination: {
-      total: 22
-    }
-  }));
+  let store = createMockStore(
+    EA([{id: 1, name: 'Walter White'}], { pagination: { total: 22 } })
+  );
 
-  var route = createRoute(['item'], {
+  let route = createRoute(['item', {
     totalPagesParam: 'pagination.total',
-    store
-  });
+  }], { store });
 
-  callModelHook(route);
+  let model = callModelHook(route);
 
-  assert.equal(route.get('_totalPages'), 22, '_totalPages');
+  assert.equal(model.get('totalPagesParam'), 'pagination.total', 'totalPagesParam');
+  assert.equal(model.get('_totalPages'), 22, '_totalPages');
 });
 
 module('RouteMixin - reaching the end', {
```

Comment:
Could split each one into its own test file and use `beforeEach` instead of setup.